case-lib/lib.sh

```diff
@@ -752,17 +752,37 @@ func_lib_check_pa()
 # However, 1. arrays would complicate the user interface 2. ALSA does not
 # seem to need arguments with whitespace or globbing characters.
 
+channel=$(func_pipeline_parse_value "$idx" channel)
+rate=$(func_pipeline_parse_value "$idx" rate)
+fmt_elem=$(func_pipeline_parse_value "$idx" fmt_elem)
+dev=$(func_pipeline_parse_value "$idx" dev)
+duration=$(func_pipeline_parse_value "$idx" duration)
+
 aplay_opts()
 {
-    dlogc "aplay $SOF_ALSA_OPTS $SOF_APLAY_OPTS $*"
-    # shellcheck disable=SC2086
-    aplay $SOF_ALSA_OPTS $SOF_APLAY_OPTS "$@"
+    if [[ $SOF_TINY_ALSA == true ]]; then
+        card_nr=$(echo "$dev" | cut -d ':' -f2 | cut -d ',' -f1)
+        dev_nr=$(echo "$dev" | cut -d ',' -f2)
+        sox -n -r "$rate" -c "$channel" noise.wav synth "$duration" white
+        tinyplay -D "$card_nr" -d "$dev_nr"  -i wav noise.wav
+    else
+        dlogc "aplay $SOF_ALSA_OPTS $SOF_APLAY_OPTS $*"
+        # shellcheck disable=SC2086
+        aplay $SOF_ALSA_OPTS $SOF_APLAY_OPTS "$@"
+    fi
 }
 arecord_opts()
 {
-    dlogc "arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS $*"
-    # shellcheck disable=SC2086
-    arecord $SOF_ALSA_OPTS $SOF_ARECORD_OPTS "$@"
+    if [[ $SOF_TINY_ALSA == true ]]; then
+        card_nr=$(echo "$dev" | cut -d ':' -f2 | cut -d ',' -f1)
+        dev_nr=$(echo "$dev" | cut -d ',' -f2)
```

Comment:
This is a copy of the code in `aplay_opts()`, move this to some new `set_tiny_alsa_params()` function.